libs/utils/android/workload.py

```diff
@@ -88,6 +88,29 @@ def run(self, out_dir, collect='',
             **kwargs):
         raise RuntimeError('Not implemeted')
 
+    def setupTargetScript(self, actions):
+        """
+        Push a script to the target
+
+        :param actions: List of actions(commands) to run
+        :type actions: list(str)
+        """
+
+        actions = ['#!{} sh'.format(self._target.busybox)] + actions
+        actions = str.join('\n', actions)
+
+        self._scriptpath = '{}/remote_script.sh'.format(self._target.executables_directory)
+
+        self._target.execute('echo \"{}\" > {}'.format(actions, self._scriptpath))
```

Comment:
When hacking similar stuff in the past I have had issues with this approach due to the length of the command (IIRC I found that ADB on Google Pixel has a 4096 byte limit), so it would be more robust to write this to a file and then use `push`. Might be slower, but I think we'll have to live with that.